crates/api/src/lib.rs

```diff
@@ -25,15 +27,23 @@ pub trait Perform {
 pub(crate) fn captcha_as_wav_base64(captcha: &Captcha) -> String {
   let letters = captcha.as_wav();
 
-  let mut concat_letters: Vec<u8> = Vec::new();
-
+  // Decode each wav file, concatenate the samples
+  let mut concat_samples: Vec<i16> = Vec::new();
+  let mut any_header: Option<wav::Header> = None;
   for letter in letters {
-    let bytes = letter.unwrap_or_default();
-    concat_letters.extend(bytes);
+    let mut cursor = Cursor::new(letter.unwrap_or_default());
+    let (header, samples) = wav::read(&mut cursor)
+        .expect("Unable to parse wav file");
+    any_header = Some(header);
+    concat_samples.extend(samples.as_sixteen().expect("Expected 16-bit samples"));
```

Comment:
You shouldnt use expect as this can crash.

Reply:
I can change it to propagate errors instead. Should I have the function return a Result, or just an empty string on failure as before?

Reply:
It should definitely return a result so that the client knows something went wrong. Checking for empty string would be very awkward.

Reply:
OK, fixed so that captcha_as_wav_base64 returns a result. I changed GetCaptcha to log the error and populate an empty string for wav, because that way it doesn't crash and break the png captcha too. In practice the client handles that gracefully and doesn't display the audio captcha in that case.

Maybe a follow-up would be to change the CaptchaResponse struct so that it's optional. Would that be worth it?